### PR TITLE
Cleanup virtctl

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

This PR cleans up some of virtctl vm create's code and test code, namely:

- vm_test: Avoid fmt.Sprintf where possible
- vm_test: Replace pointer with ptr
- vm_test: Cleanup volume-import tests a little
- virtctl: Drop unused VM param from volumeImportFn
- virtctl: Add missing validation of VDDK params

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 
-

### Special notes for your reviewer

Merge after #11144 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by 0xFelix** [comment]
/cc @oshoval 
/cc @alicefr 

Can you PTAL?

**Comment by oshoval** [comment]
Nice thanks

We can consider adding staticcheck or equivalents (on a followup),
it can detect for example the deprecated pointer lib and possibly more issues.

nit: 3rd commit message has a typo `volum`

PR desc, Fixes # can be removed if desired, and also the special notes as the mentioned PR is merged.

Btw about the first commit that inline the consts, since it is a VM it is fine, but if it was VMI, imho
it is better to have a const in one place because it might need to be changed in the future.

/lgtm

**Comment by 0xFelix** [comment]
@oshoval Fixed typo and rebased on main.

**Comment by oshoval** [comment]
Thanks
/lgtm

**Comment by alicefr** [comment]
Everything looks great!
/approve

File: pkg/virtctl/create/vm/vm_test.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 
 	k8sv1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
-	"k8s.io/utils/pointer"
+	"k8s.io/utils/ptr"
```
[the remark this answers is not in the document]
Comment (oshoval) [reply pkg/virtctl/create/vm/vm_test.go:12]: yes, but please see this
https://github.com/kubevirt/kubevirt/pull/11188#discussion_r1485721039

[the remark this answers is not in the document]
Comment (EdDev) [reply pkg/virtctl/create/vm/vm_test.go:12]: I do not think you should have used this 3rd party package, we have an internal one which is one-liner.

But this topic is not worth blocking this change.

**Pull request merged**